### PR TITLE
`lib.path.splitStorePath`: init

### PR DESCRIPTION
**Pull request by infinisil**
> [!Note]
> This was dependent on https://github.com/NixOS/nixpkgs/pull/273883

## Description of changes

In order to implement https://github.com/NixOS/nixpkgs/issues/269283 I need a function to split a path within a store path into its store path and the subpath. So this PR adds a function to do that:
```nix
splitStorePath /nix/store/nvl9ic0pj1fpyln3zaqrf4cclbqdfn1j-foo/bar/baz
-> {
  storePath = /nix/store/nvl9ic0pj1fpyln3zaqrf4cclbqdfn1j-foo;
  subpath = "./bar/baz";
}
```

This work is sponsored by [Antithesis](https://antithesis.com/) :sparkles:

## Things done
- [x] Documentation
- [x] Tidy code
- [x] Tests

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by fricklerhandwerk — APPROVED** [review]

**Comment by infinisil** [comment]
I originally thought this function was necessary for https://github.com/NixOS/nixpkgs/issues/269283, but turns out it's not. While this would be ready, we shouldn't merge this if there's no use case for it. So I'll just close for now until perhaps another use case comes up.

**Pull request closed without merging**
